build_scripts/changed_functions.py

```diff
@@ -20,21 +20,20 @@ def parse_args():
 
 def main():
     args = parse_args()
-    print(f"Got function folders: {args.folders!r}")  # noqa
-    print(f"And deploy all folders: {args.deploy_all!r}")  # noqa
+    print(f"Got function folders: {args.folders!r}")
+    print(f"And deploy all folders: {args.deploy_all!r}")
     function_folders = {f.strip() for f in args.folders[0].split(",")}
-    deploy_all_folders = (args.deploy_all or {}) and {f.strip() for f in args.deploy_all.split(",")}
-
-    process = subprocess.Popen("git diff --name-only HEAD^ HEAD".split(), stdout=subprocess.PIPE)  # nosec
-    changed_folders, _ = process.communicate()
-    changed_folders = list(
-        {
-            "/".join(clean.parts if clean.name != "schedules" else clean.parts.parts)
-            for f in changed_folders.decode().split("\n")
-            if (clean := Path(f).parent) and clean.exists()
-        }
-    )
-    print(f"Detected changed folders: {changed_folders}")  # noqa
+    deploy_all_folders = set()
+    if args.deploy_all:
+        deploy_all_folders = {f.strip() for f in args.deploy_all.split(",")}
```

Comment:
Sorry for rewriting this, but I just hate using `and` outside of boolean evaluations 😂 (`or` I'm just fine with though for some reason...)

Reply:
I do not mind, if you find it more readable you are probably not alone.